ha/util/consul_kv_store.py

```diff
@@ -189,5 +270,80 @@ def get_keys(self, prefix):
             List of keys
         """
         if not ConsulKvStore._keys.get(prefix):
+
+            # get cached keys if batch put is enabled
+            payload_keys = None
+            if self._enable_batch_put:
+                # Using dictionary comprehension + startswith()
+                # Prefix key match in dictionary
+                payload_keys = [key for key in self._payload if key.startswith(prefix)]
+
             ConsulKvStore._keys[prefix] = self._consul.kv.get(prefix, keys=True)[1]
+
+            # Merge lists with unique values
+            if payload_keys:
+                ConsulKvStore._keys[prefix] = list(set(payload_keys + ConsulKvStore._keys[prefix]))
```

Comment:
What happens if keys with this prefix are present in both cache and consul ? 
ex prefix is abcd>pqr
Consul conaints :  abcd>pqr>ertg>dkljg
cache contains: abcd>pqr>pohg

both will not be returned here

Reply:
no, both will be returned